### DISCUSSION TITLE
wrong server declaration

### DISCUSSION THREAD

**Discussion by danielRuoss** [oshi/oshi#2223] [Q&A]
using oshi to determine the windows version 
operation system from system.operatingSystem.toString()  is : Microsoft Windows Server 2019 build 20348
but build 20348 is Server 2022 according to https://support.microsoft.com/en-au/topic/may-10-2022-kb5013944-os-build-20348-707-05509703-187a-4d5b-97f5-8793dbb22991

**Comment by dbwiddis** [comment]
OSHI just pulls the value from the `os.name` system property, which comes from the JDK.

[Latest JDK main branch correctly shows](https://github.com/openjdk/jdk/blob/af2de975cbc46ad91c736b110a2212cd32ba60a7/src/java.base/windows/native/libjava/java_props_md.c#L559-L560) builds > 20347 as Server 2022.

That code was changed in [September 2021](https://github.com/openjdk/jdk/commit/c2e015c3c1a2274112bb8e6671a85bc7fb624fde) so I would think any recent JDK would have it backported.

We could put in a manual hack to do the same thing in OSHI (we do it for Windows 10 --> 11):
```
            if ("10".equals(version) && buildNumber.compareTo("22000") >= 0) {
                this.version = "11";
            } else {
                this.version = version;
            }
```
Or you can just update your JDK :)

**Reply by danielRuoss** [reply]
ah okay thanks for the fast replay i will update the jdk ;)

**Reply by dbwiddis** [reply]
I added #2224 to fix it in the next release anyway, since I know a lot of server-based monitoring systems are way behind on JDK updates :)
